frontend/src/metabase/components/ItemsTable/BaseItemsTable.styled.tsx

```diff
@@ -6,81 +6,61 @@ import Link from "metabase/core/components/Link";
 import AdminS from "metabase/css/admin.module.css";
 import { color } from "metabase/lib/colors";
 import BaseModelDetailLink from "metabase/models/components/ModelDetailLink";
-import {
-  breakpointMaxMedium,
-  breakpointMinLarge,
-} from "metabase/styled-components/theme/media-queries";
-import { Icon } from "metabase/ui";
+import { FixedSizeIcon } from "metabase/ui";
 
-const LAST_EDITED_BY_INDEX = 3;
-const LAST_EDITED_AT_INDEX = 4;
+import type { ResponsiveProps } from "./utils";
+import { getContainerQuery } from "./utils";
 
-export const Table = styled.table<{ canSelect: boolean }>`
+export const Table = styled.table<{ isInDragLayer?: boolean }>`
   background-color: ${color("white")};
   table-layout: fixed;
   border-collapse: unset;
-  border-radius: 8px;
+  border-radius: 0.5rem;
+  overflow: hidden;
 
   thead {
     th {
       border-top: 1px solid ${color("border")};
 
       &:first-of-type {
-        border-top-left-radius: 8px;
-        border-left: 1px solid ${color("border")};
+        border-start-start-radius: 8px;
+        border-inline-start: 1px solid ${color("border")};
       }
 
       &:last-child {
-        border-top-right-radius: 8px;
-        border-right: 1px solid ${color("border")};
+        border-start-end-radius: 8px;
+        border-inline-end: 1px solid ${color("border")};
       }
     }
   }
 
-  ${props => {
-    const offset = props.canSelect ? 1 : 0;
-    const offsetEditedByIndex = LAST_EDITED_BY_INDEX + offset;
-    const offsetEditedAtIndex = LAST_EDITED_AT_INDEX + offset;
-
-    return `
-      ${breakpointMaxMedium} {
-        & td:nth-of-type(${offsetEditedByIndex}),
-        th:nth-of-type(${offsetEditedByIndex}),
-        col:nth-of-type(${offsetEditedByIndex}),
-        td:nth-of-type(${offsetEditedAtIndex}),
-        th:nth-of-type(${offsetEditedAtIndex}),
-        col:nth-of-type(${offsetEditedAtIndex}) {
-          display: none;
-        }
-      }
-    `;
-  }}
```

Comment:
Previously, columns were made responsive to viewport width changes through these hard-coded indexes. On this branch, the same effect is achieved with the help of `@container` queries